src/firebolt/async_db/cursor.py

```diff
@@ -366,7 +387,7 @@ async def _validate_set_parameter(self, parameter: SetParameter) -> None:
         # set parameter passed validation
         self._set_parameters[parameter.name] = parameter.value
 
-    async def _do_execute(
+    async def _do_execute(  # noqa: C901
```

Comment:
I think we shouldn't just blanket ignore warnings when they pop up. If a function is becoming complex we'd better refactor it. SonarCloud is also triggering on this, but not failing.

Reply:
Done.